pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/QueryQuotaClusterIntegrationTest.java

```diff
@@ -234,30 +235,44 @@ public void testApplicationAndDatabaseQueryQuotaWithTableQueryQuotaWithExtraBrok
    * Then runs the query load with double the max rate and expects queries to fail due to quota breach.
    * @param maxRate max rate allowed by the quota
    */
-  void testQueryRate(float maxRate)
-      throws Exception {
+  void testQueryRate(int maxRate) {
     verifyQuotaUpdate(maxRate);
     runQueries(maxRate, false);
     //increase the qps and some of the queries should be throttled.
     runQueries(maxRate * 2, true);
   }
 
-  void testQueryRateOnBroker(float maxRate)
-      throws Exception {
+  void testQueryRateOnBroker(float maxRate) {
     verifyQuotaUpdate(maxRate);
     runQueriesOnBroker(maxRate, false);
     //increase the qps and some of the queries should be throttled.
     runQueriesOnBroker(maxRate * 2, true);
   }
 
+  private static void sleep(long end, double iterationsLeft) {
```

Comment:
I don't fully follow the change. Seems we are getting interruption during sleep? Is this expected?

Reply:
That InterruptedException catch is just in case something could interrupt the waiting thread. 
The main fix is about adjusting the sleep duration vs time left to reach qps deadline.

Reply:
I'd suggest not swallow `InterruptedException` as it is unexpected